scripts/src/chartprreview/chartprreview.py

```diff
@@ -339,7 +358,7 @@ def main():
         report_info_path = ""
         if report_generated and report_generated == "True":
             match_checksum(args.directory,generated_report_info_path, category, organization, chart, version)
-        else:
+        elif not provider_delivery or provider_delivery == "False":
```

Comment:
Yes and yes, this logic is running when a report was submitted in the PR. In this case we may have also generated a report to match checksum, but in provider delivery this would not be the case. Before at this stage we would check for a valid chart url in the report, but for provider delivery control a the chart url will not be set.